code/go/0chain.net/blobbercore/filestore/state.go

```diff
@@ -0,0 +1,208 @@
+package filestore
+
+import (
+	"context"
+	"errors"
+	"sync"
+	"time"
+
+	"github.com/0chain/blobber/code/go/0chain.net/blobbercore/datastore"
+	"github.com/0chain/blobber/code/go/0chain.net/core/common"
+	"github.com/0chain/blobber/code/go/0chain.net/core/logging"
+	"go.uber.org/zap"
+)
+
+type allocation struct {
+	mu            *sync.Mutex
+	allocatedSize uint64
+	filesNumber   uint64
+	filesSize     uint64
+
+	tmpMU       *sync.Mutex
+	tmpFileSize uint64
+}
+
+func (fs *FileStore) setAllocation(allocID string, alloc *allocation) {
+	fs.rwMU.Lock()
+	defer fs.rwMU.Unlock()
+	fs.mAllocs[allocID] = alloc
+}
+
+func (fs *FileStore) getAllocation(allocID string) *allocation {
+	fs.rwMU.RLock()
+	defer fs.rwMU.RUnlock()
+	return fs.mAllocs[allocID]
+}
+
+func (fs *FileStore) removeAllocation(ID string) {
+	fs.rwMU.Lock()
+	defer fs.rwMU.Unlock()
+	delete(fs.mAllocs, ID)
+}
+
+func (fs *FileStore) initMap() error {
+	ctx, cnCl := context.WithCancel(context.Background())
+	defer cnCl()
+
+	ctx = datastore.GetStore().CreateTransaction(ctx)
+	db := datastore.GetStore().GetTransaction(ctx)
+	if db == nil {
+		return errors.New("could not get db client")
+	}
+
+	var allocations []*dbAllocation
+	if err := db.Model(&dbAllocation{}).Find(&allocations).Error; err != nil {
```

Comment:
Will this load all the allocations?

Reply:
Yes it will.
Do you mean we should get allocations in batches?

Reply:
Just worried about the performance when we have large number of allocations, so as the RAM resources it would use. Perhaps do it in batches